app/controllers/api/v1/concepts_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Api::V1::ConceptsController < Api::ApiController
+  doorkeeper_for :create
+
+  def create
+    concept = Concept.new(concept_params)
+    if concept.save
+      render json: {concept: Api::SimpleSerializer.new(concept).as_json(root: false)}
+    else
+      render json: concept.errors, status: 422
+    end
+  end
+
+  def index
+    render json: Concept.all_with_depth
+  end
+
+  private
+
+  def concept_params
+    parent = Concept.find_by uid: params[:parent_uid]
+    params[:parent_id] = parent.id
+    params.permit(:name, :parent_id)
+  end
+
```

Comment:
@marcellosachs A more idiomatic way to do this would be to add a `parent_uid=` method to the Concept model and then allow `:parent_uid` directly in the strong params list. The `parent_uid=` method would set the parent_id value by looking it up via the uid, similar to how you have it in the controller.